plugins/modules/host_hide.py

```diff
@@ -0,0 +1,219 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+
+# Copyright: (c) 2023, CrowdStrike Inc.
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+from __future__ import absolute_import, division, print_function
+
+__metaclass__ = type
+
+DOCUMENTATION = r"""
+---
+module: host_hide
+
+short_description: Hide/Unhide hosts from the Falcon console
+
+version_added: "4.0.0"
+
+description:
+  - Manages the visibility of hosts in the Falcon console.
+  - To prevent unnecessary detections from an inactive or a duplicate host,
+    you can opt to hide the host from the console. This action does not uninstall or
+    deactivate the sensor. Detection reporting resumes after a host is unhidden.
+  - The module will return a list of successfull and failed hosts IDs for the action performed.
```

Comment:
Overall I really don't like this "action module" approach. I'd expect something much more like this in Ansible:

```yaml
- host:
    ids:
      - abcd1234
    state: hidden # or active
```

In this case the `host` module would probably grow pretty large over time and be one of the primary modules. E.g. control sensor update policies attached to a host with a `sensor_update_policy` attribute. But that seems ok to me since hosts are a central part of the platform.

Reply:
Host module would imply being able to manage the full lifecycle of an object (ie host). But that is not how the API works, we can't create hosts, or delete them, so basically it makes more sense imo to break down the hosts api into actionable modules. Ie, host_hide, host_contain, host_info

Reply:
Maybe something in the middle to make it a little more idempotent-y:

```yaml
- host_hide:
    hosts:
      - abcd1234
    # hidden: true is the default
```

```yaml
- host_hide:
    hosts:
      - abcd1234
    hidden: false
```

Same thing for `host_contain` with a `contained` boolean.